ui/ducks/confirm-transaction/confirm-transaction.duck.js

```diff
@@ -118,8 +130,32 @@ export default function reducer(state = initState, action = {}) {
         ...state,
         nonce: action.payload,
       };
-    case CLEAR_CONFIRM_TRANSACTION:
-      return initState;
+    case SET_DETAILS_FOR_CONFIRM_TX: {
+      return {
+        ...state,
+        sendTxDetailPerId: {
+          ...state.sendTxDetailPerId,
+          [action.payload.transactionId]: {
+            amountMode: action.payload.amountMode,
+            assetType: action.payload.assetType,
+          },
+        },
+      };
+    }
+    case CLEAR_CONFIRM_TRANSACTION: {
+      // Preserve sendTxDetailPerId
+      return {
+        ...initState,
+        sendTxDetailPerId: state.sendTxDetailPerId,
+      };
+    }
+    case actionConstants.COMPLETED_TX: {
```

Comment:
If we use the `boolean` as mentioned in other comments, do we even need to clear our state since at worst it will just be a collection of booleans for some previously approved or rejected transactions?

Reply:
My concern was somehow this part is persisted in the redux, but tested that it's not, so removed.